gnd/src/main/java/com/google/android/gnd/ui/editrecord/EditRecordViewModel.java

```diff
@@ -78,6 +86,47 @@
     this.showUnsavedChangesDialogEvents = new SingleLiveEvent<>();
     this.showErrorDialogEvents = new SingleLiveEvent<>();
     this.authManager = authenticationManager;
+    this.editRecordRequests = PublishSubject.create();
+    this.recordSaveRequests = PublishSubject.create();
+
+    disposeOnClear(
+            recordSaveRequests.zipWith(authManager.getUser(), this::saveRecord)
+            .switchMap(Result::wrapObservable)
```

Comment:
What is this line trying to do? If there's no substream from which to catch errors, why, wrap in Result?

Reply:
`saveRecord` returns an Observable, so we need to call `switchMap` to flatten the streams. We essentially have two options, call `switchMap` with the identity function `r -> r` or call it with `Result::wrapObservable`. I chose the latter because it gives us a standardized way to handle errors. If the observable returned by `saveRecord` throws an error, it will instead return `Result::error` and continue emitting. I don't think this adds too much in this case, though, so I've switched to the more direct approach of using the identity function.